common/txmgr/types/client.go

```diff
@@ -3,38 +3,40 @@ package types
 import (
 	"context"
 	"math/big"
+
+	"github.com/smartcontractkit/chainlink/v2/common/types"
 )
 
 // A generic client interface for communication with the RPC node
 // Every native chain must implement independently
-type Client[CHAINID any, ADDR comparable, BLOCK any, TX any, HASH comparable, TXRECEIPT any, EVENT any, EVENTOPS any] interface {
+type Client[CHAINID ID, SEQ Sequence, ADDR types.Hashable[ADDR], BLOCK any, TX any, HASH types.Hashable[HASH], TXRECEIPT any, EVENT any, EVENTOPS any] interface {
 	// ChainID stored for quick access
 	ConfiguredChainID() CHAINID
 	// ChainID RPC call
 	ChainID() (CHAINID, error)
 
-	Accounts[ADDR]
+	Accounts[ADDR, SEQ]
 	Transactions[TX, HASH, TXRECEIPT]
 	Events[EVENT, EVENTOPS]
 	Blocks[BLOCK, HASH]
 
 	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
 }
 
-type Accounts[ADDR comparable] interface {
+type Accounts[ADDR types.Hashable[ADDR], SEQ Sequence] interface {
 	BalanceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (*big.Int, error)
 	TokenBalance(ctx context.Context, accountAddress ADDR, tokenAddress ADDR) (*big.Int, error)
-	SequenceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (uint64, error)
+	SequenceAt(ctx context.Context, accountAddress ADDR, blockNumber *big.Int) (SEQ, error)
```

Comment:
The return type imo should be (*SEQ, error)

Reply:
SEQ is int64 after the latest change, so I'm keeping it as it is